.travis.yml

```diff
@@ -1,56 +1,73 @@
-            stage: build
-            if: NOT type = cron
-            name: Build docker image
+            stage: test
+            name: ECS
             script:
-                - echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin
-                - docker pull pehapkari/pehapkari.cz || true
-                - docker build --cache-from pehapkari/pehapkari.cz -t pehapkari/pehapkari.cz:$TRAVIS_COMMIT --target production -f Dockerfile .
-                - docker push pehapkari/pehapkari.cz
+                - composer check-cs
 
-        # 2. using built Docker image from previous step
         -
-            stage: test
-            name: Static Analysis
+            name: PHPStan
             script:
-                - docker run --rm pehapkari/pehapkari.cz:$TRAVIS_COMMIT composer check-cs
-                - docker run --rm pehapkari/pehapkari.cz:$TRAVIS_COMMIT composer phpstan
+                - composer phpstan
 
         -
             name: PHPUnit
-            script: docker run --rm pehapkari/pehapkari.cz:$TRAVIS_COMMIT php vendor/bin/phpunit
+            script:
+                - vendor/bin/phpunit
 
         -
             name: Rector
-            script: docker run --rm pehapkari/pehapkari.cz:$TRAVIS_COMMIT composer rector
+            script:
+                - composer rector-dry
 
         -
             name: Linting YAML and Templates
             script:
-                - docker run --rm pehapkari/pehapkari.cz:$TRAVIS_COMMIT bin/console lint:yaml src packages
-                - docker run --rm pehapkari/pehapkari.cz:$TRAVIS_COMMIT bin/console lint:twig packages templates
+                - bin/console lint:yaml src packages
+                - bin/console lint:twig packages templates
 
         -
             name: Dead links
             script: docker run --rm pehapkari/pehapkari.cz:$TRAVIS_COMMIT composer dead-links
 
+        # 2. Build Docker image → deploy
         -
-            name: Prod cache warmup
-            script: docker run --rm -e APP_ENV=prod pehapkari/pehapkari.cz:$TRAVIS_COMMIT bin/console cache:warmup
+            stage: build
+            name: 'Build Docker Image'
+            if: NOT type = cron
+            install:
+                - echo "composer install is run inside the Docker image"
+            script:
+                - echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin
+                - docker pull pehapkari/pehapkari.cz || true
+                - docker build --cache-from pehapkari/pehapkari.cz -t pehapkari/pehapkari.cz:$TRAVIS_COMMIT --target production -f Dockerfile .
+                - docker push pehapkari/pehapkari.cz
+
+        -
+            stage: wramup
+            name: 'Production Cache Warmup'
+            install:
+                - echo "composer install is run inside the Docker image"
+            script:
+                - docker run --rm -e APP_ENV=prod pehapkari/pehapkari.cz:$TRAVIS_COMMIT bin/console cache:warmup
```

Comment:
I bet this will not work, because image `pehapkari/pehapkari.cz` with tag `$TRAVIS_COMMIT` will not be available here, but maybe i am mistaken.

Reply:
Give me a sec before reviews, I have to try/learn it first :)

Reply:
Yaay, im too fast, sorry :smile: sure